docs/enhancements/1-crds-and-controllers.md

```diff
@@ -0,0 +1,430 @@
+
+All resources will contain [standard conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) in their status.
+
+#### MeshFederation
+
+`MeshFederation` includes settings related only to mesh federation, like remote ingress address, identities, etc.
+This resource must be cluster-scoped, because it will be used as an owner for `ImportedService` resources, which can be created in multiple namespaces.
+
+I would like to move away from thinking about "peers", as this is just an implementation detail.
+I also thought that it may be confusing why peer address is used as the address for remote services.
+
+Resource structure:
+
+```yaml
+apiGroup: federation.openshift-service-mesh.io/v1alpha1
+kind: MeshFederation
+metadata:
+  name: default
+spec:
+  local:
+    # ID is a unique identifier of the FDS peer, and it's used as a suffix in its Service and ServiceAccount names.
+    id: east
+    # Network name used by Istio for load balancing.
+    network: east
+    # Optional.
+    # Trust domain is a property of the identity, which can't be determined by the controller.
+    # Service account and namespace will be obtained from the downward API.
+    # Default value: "cluster.local".
+    trustDomain: mesh.east
+    # Optional.
+    # If no ingress is specified, it means the controller supports only single network topology.
+    ingress:
+      # Optional.
+      # Local ingress type specifies how to expose exported services.
+      # Currently, only two types are supported: istio and openshift-router.
+      # If "istio" is set, then the controller assumes that the Service associated with federation ingress gateway
+      # is LoadBalancer or NodePort and is directly accessible for remote peers, and then it only creates
+      # an auto-passthrough Gateway to expose exported Services.
+      # When "openshift-router" is enabled, then the controller creates also OpenShift Routes and applies EnvoyFilters
+      # to customize the SNI filter in the auto-passthrough Gateway, because the default SNI DNAT format used by Istio
+      # is not supported by OpenShift Router.
+      type: istio
+      # Optional.
+      # Can be empty if ingress.type is openshift-router.
+      gateway:
+        # Ingress gateway selector specifies to which workloads Gateway configurations will be applied.
+        selector:
+          app: federation-ingress-gateway
+        # Optional.
+        port:
+          # Optional.
+          # Port name of the ingress gateway Service.
+          # This is relevant only when the ingressType is openshift-router, but it cannot be empty.
+          name: tls-passthrough
+          # Optional.
+          # Port of the ingress gateway Service.
+          number: 15443
+    # Optional.
+    controlPlane:
+      # Optional.
+      # Control plane namespace used to create mesh-wide resources.
+      namespace: istio-system
```

Comment:
@jewertow If our controller will always be deployed in the same namespace as Istio is, I think we don't need this field at all. We can always get it from pod's env vars.